### ISSUE TITLE
String containing an "e" is now parsed as an exponential expression in 3.12.0

### ISSUE THREAD

**Issue by SkxNMDI5** [nodeca/js-yaml#431]
In an empty directory :

```
npm install js-yaml@3.6.1
node -e 'const yml = require("js-yaml"); let obj = yml.safeLoad("test: 2e30"); console.log(obj);'
{ test: '2e30' }

npm install js-yaml@3.12.0
node -e 'const yml = require("js-yaml"); let obj = yml.safeLoad("test: 2e30"); console.log(obj);'
{ test: 2e+30 }
```

This change in behavior can have major impacts (for my part when using the serverless framework which has dependency on "js-yaml": "^3.6.1")

**Comment by puzrin** [comment]
Is this issue related anyhow to invalid yaml spec implementation? Did you inspected spec prior to create?

If not - it will be closed, because your input should be properly quoted, independently of this package bugs.

**Comment by puzrin** [comment]
http://nodeca.github.io/js-yaml/#yaml=cXVvdGVkOiAnMi4zZTQnCnVucXVvdGVkOiAyLjNlNAo=

Everything works as expected.

Closing - can not understand goal of this issue.

**Comment by SkxNMDI5** [comment]
Funny 
which version of js-yaml is executed behind http://nodeca.github.io/js-yaml/ ?

**Comment by puzrin** [comment]
Always latest.

> The goal is simply that between 3.6.1 and 3.12.0 there is a breaking change.

As i explained before, if you have problem with your yaml data, that does not mean that change is breaking. If you come with such statements, you need solid proofs. For example, that package violates yaml spec. If package fixed implementation - that's a bugfix, not a break.

**Comment by SkxNMDI5** [comment]
I understand.

In my environment I have a different behavior but can't find why.
I have tried :

 - in the local environment of my workstation (node  v9.8.0)
 - in a clean environnement of a docker container with node versions @latest (v10.6.0) and 8.11.3-alpine.

But the quoted string is always parsed as a scientific float.

```
$ npm install js-yaml@3.12.0
$ node -e 'const yml = require("js-yaml"); let obj = yml.safeLoad("quoted: '2.3e4'\nunquoted: 2.3e4"); console.log(obj);'
{ quoted: 23000, unquoted: 23000 }
```

So I decided to switch to the the tag notation (!!str) of the specification  that js-yaml respects . 
So that will be fine in my case.

Anyway thank you for maintaining this project.
